### PR TITLE
#31 Remove controlName from the metrics

### PR DESCRIPTION
**Pull request by TerezaHrubesova**

### CODE REVIEW

**Review by Zejnilovic — DISMISSED** [review]

File: examples-s3-sdk-extension/src/main/scala/za/co/absa/atum/examples/SampleSdkS3Measurements2.scala [hunk 1]
```diff
@@ -89,7 +90,7 @@ object SampleSdkS3Measurements2 {
 
   private def extractCheckpointsRecordCounts(controlMeasure: ControlMeasure): Seq[(String, Int)] = {
     controlMeasure.checkpoints.map { checkpoint =>
-      val count: Int = checkpoint.controls.collectFirst { case Measurement("recordCount", _, _, value) => value.toString.toInt }.getOrElse(0)
+      val count: Int = checkpoint.controls.collectFirst { case Measurement(ControlType.Count.value, _, value) => value.toString.toInt }.getOrElse(0)
```
Comment (Zejnilovic) [inline examples-s3-sdk-extension/src/main/scala/za/co/absa/atum/examples/SampleSdkS3Measurements2.scala:93]: @dk1844 could you please look at this

Comment (dk1844) [reply examples-s3-sdk-extension/src/main/scala/za/co/absa/atum/examples/SampleSdkS3Measurements2.scala:93]: In what way to look at this? In general, I don't mind this change.

_Deep-dive:_
The change is probably based on the fact that the example code expects to find an _INFO file that contains certain counts (such file may be in fact identical or similar to file `examples/data/input/wikidata.csv.info` - this file specifically has the first control count of type `ControlType.Count`). If you have all the prerequisites (locally defined env variables and a valid SAML profile, it is also manually runnable).
In the end, the change specifically decided to assume a control type.

tldr: the example is thus a bit different, but equally valid. 🤷‍♂️

**Review by dk1844 — DISMISSED** [review]
LGTM.

The nice thing about this change is that it seems to be backward-compatible (i.e. the extra existing field should not break things)

**Review by benedeki — COMMENTED** [review]
Really minor things. Just some "nice code" stuff.

File: atum/src/test/scala/za/co/absa/atum/persistence/TestResources.scala [hunk 2]
```diff
@@ -27,7 +27,7 @@ object TestResources {
       ControlMeasureMetadata("AtumTest", "CZ", "Snapshot", "example_input.csv", "public", 1, "01-01-2020", Map.empty),
```
[the remark this answers is not in the document]
Comment (TerezaHrubesova) [reply atum/src/test/scala/za/co/absa/atum/persistence/TestResources.scala:27]: Type added.

File: examples-s3-sdk-extension/src/main/scala/za/co/absa/atum/examples/SampleSdkS3Measurements2.scala (hunk 1 above)
Comment (TerezaHrubesova) [reply examples-s3-sdk-extension/src/main/scala/za/co/absa/atum/examples/SampleSdkS3Measurements2.scala]: I added "*" as controlCol. Taken from wikidata.csv.info.

File: atum/src/test/scala/za/co/absa/atum/ControlMeasurementsSpec.scala [hunk 3]
```diff
@@ -39,55 +39,46 @@ class ControlMeasurementsSpec extends AnyFlatSpec with Matchers with SparkTestBa
 
   val measurementsIntOferflow = List(
```
[the remark this answers is not in the document]
Comment (TerezaHrubesova) [reply atum/src/test/scala/za/co/absa/atum/ControlMeasurementsSpec.scala]: Changed to private.

File: atum/src/test/scala/za/co/absa/atum/utils/controlmeasure/ControlMeasureUtilsSpec.scala [hunk 4]
```diff
@@ -38,8 +38,8 @@ class ControlMeasureUtilsSpec extends AnyFlatSpec with ControlMeasureBaseTestSui
        |"sourceType":"Source","version":1,"informationDate":"$testingDate","additionalInfo":{}},
```
[the remark this answers is not in the document]
Comment (TerezaHrubesova) [reply atum/src/test/scala/za/co/absa/atum/utils/controlmeasure/ControlMeasureUtilsSpec.scala:38]: Type annotation added.

**Review by benedeki — APPROVED** [review]
